File: app/src/main/java/code/name/monkey/retromusic/mvp/contract/AlbumContract.java
package code.name.monkey.retromusic.mvp.contract;

import code.name.monkey.retromusic.model.Album;
import code.name.monkey.retromusic.mvp.BasePresenter;
import code.name.monkey.retromusic.mvp.BaseView;
import java.util.ArrayList;

public interface AlbumContract {

  interface AlbumView extends BaseView<ArrayList<Album>> {

  }

  interface Presenter extends BasePresenter<AlbumView> {

    void loadAlbums();
  }

}

File: app/src/main/java/code/name/monkey/retromusic/mvp/contract/AlbumContract.kt
package code.name.monkey.retromusic.mvp.contract

import code.name.monkey.retromusic.model.Album
import code.name.monkey.retromusic.mvp.BasePresenter
import code.name.monkey.retromusic.mvp.BaseView
import java.util.ArrayList

interface AlbumContract {

    interface AlbumView : BaseView<ArrayList<Album>>

    interface Presenter : BasePresenter<AlbumView> {

        fun loadAlbums()
    }

}
